stubs/protobuf/google/protobuf/internal/type_checkers.pyi

```diff
@@ -0,0 +1,15 @@
+from typing import Generic, Protocol, TypeVar
+
+_T = TypeVar("_T")
+
+class _ValueChecker(Protocol[_T]):
+    def CheckValue(self, proposed_value: _T) -> _T: ...
+    def DefaultValue(self) -> _T: ...
+
+class TypeChecker(Generic[_T]):
+    def __init__(self, *acceptable_types: _T): ...
+    def CheckValue(self, proposed_value: _T) -> _T: ...
+
+class TypeCheckerWithDefault(TypeChecker[_T]):
+    def __init__(self, default_value: _T, *acceptable_types: _T): ...
+    def DefaultValue(self) -> _T: ...
```

Comment:
Ideally we could assert that TypeCheckerWithDefault[_T] implements the protocol here.
I know there was some discussion on typing-sig to have this be supported in typecheckers

For now I checked manually.